controllers/networking-calico/charts/internal/calico/templates/calico.yaml

```diff
@@ -334,43 +384,120 @@ spec:
             value: "true"
           # Uncomment these lines to enable prometheus metrics.  Since Typha is host-networked,
           # this opens a port on the host, which may need to be secured.
-          #- name: TYPHA_PROMETHEUSMETRICSENABLED
-          #  value: "true"
-          #- name: TYPHA_PROMETHEUSMETRICSPORT
-          #  value: "9093"
+#          - name: TYPHA_PROMETHEUSMETRICSENABLED
+#            value: "true"
+#          - name: TYPHA_PROMETHEUSMETRICSPORT
+#            value: "9093"
         livenessProbe:
-          exec:
-            command:
-            - calico-typha
-            - check
-            - liveness
+          httpGet:
+            path: /liveness
+            port: 9098
+            host: localhost
           periodSeconds: 30
           initialDelaySeconds: 30
         readinessProbe:
-          exec:
-            command:
-            - calico-typha
-            - check
-            - readiness
+          httpGet:
+            path: /readiness
+            port: 9098
+            host: localhost
           periodSeconds: 10
-
-
 ---
-
+apiVersion: {{ include "deploymentversion" . }}
+kind: Deployment
+metadata:
+  name: calico-typha-horizontal-autoscaler
+  namespace: kube-system
+  annotations:
+    checksum/configmap-calico: {{ include (print $.Template.BasePath "/config.yaml") . | sha256sum }}
```

Comment:
Why is this here? No container inside this deployment mounts the configmap. Also, the annotation would have needed to be part of the pod template and not part of the deployment metadata.